src/intrange.d

```diff
@@ -292,9 +292,10 @@ struct IntRange
         uinteger_t mask = type.sizemask();
         auto lower = SignExtendedNumber(0);
         auto upper = SignExtendedNumber(mask);
-        if (type.toBasetype().ty == Tdchar)
-            upper.value = 0x10FFFFUL;
-        else if (!isUnsigned)
+        /* Although dchar.max is officially 0x10FFFF, do *NOT* initialize upper to this value.
+           Doing so will cause attempts to guard against invalid out-of-range dchar values to
+           be considered unreachable. */
```

Comment:
Yes. Raising `dchar.max` would be a massive breaking change, and is not needed for my purposes.

All I did is remove the special casing for `dchar` in the VRP code, so this fix will remain correct even if someone else gets permission to actually change `dchar.max` later.